sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+"""
+Sequential feature selection
+
+"""
+
+# Author: Sebastian Raschka <se.raschka@gmail.com>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from itertools import combinations
+from .base import SelectorMixin
+from ..base import BaseEstimator
+from ..base import MetaEstimatorMixin
+from ..base import clone
+from ..utils.validation import check_is_fitted
+from ..model_selection import cross_val_score
+from ..metrics.scorer import check_scoring
+
+
+class SequentialFeatureSelector(BaseEstimator, MetaEstimatorMixin,
+                                SelectorMixin):
+    """Feature selector that selects features via greedy search.
+
+    This Sequential Feature Selector adds (forward selection) or
+    removes (backward selection) the features (X) to form a feature subset
+    in a greedy fashion that optimizes the extrinsic performance metric
+    of a Regressor or Classifier on the desired ouputs (y).
+
+    Read more in the :ref:`User Guide <sequential_feature_selection>`.
+
+    Parameters
+    ----------
+    estimator : scikit-learn Classifier or Regressor
+        Invoking the ``fit`` method on the `SequentialFeatureSelector`
+        will fit a clone of the original estimator that
+        will be stored in the class attribute `self.estimators_`.
+
+    n_features_to_select : int or tuple (default=1)
+        An integer arguments specifies the number of features to select,
+        where n_features_to_select < the full feature set.
+        Optionally, a tuple containing a min and max value can be provided
+        so that the feature selector will return a feature subset between
+        with min <= n_features <= max that scored highest in the evaluation.
+        For example, the tuple (1, 4) will return any combination from
+        1 up to 4 features instead of a fixed number
+        of features `n_features_to_select`.
+
+    scoring : str, callable, or None (default=None)
+        A string (see model evaluation documentation) or a scorer
+        callable object / function with signature `scorer(estimator, X, y)`.
+
+    forward : bool (default=True)
+        Performs forward selection if True and backward selection, otherwise.
```

Comment:
Is there a heuristic about which to use when that we should implement? Or because they're greedy, they're likely to land up finding quite different solutions (in which case, are you sure they belong in one class?)

Reply:
Yeah, they can yield different solutions. In practice, I would say it depends on the dataset. If you have e.g., >10000 features and want to select 10 features, `forward=False` would be far more expensive than `forward=True`. I think it would be good to keep both options, describe the pros and cons in the docs, and let the users decide.